app/src/main/kotlin/PushYourLuck/DiceClass.kt

```diff
@@ -0,0 +1,55 @@
+package PushYourLuck
+
+import StatelessSolver
+import kotlin.random.Random
+
+class DiceClass(val nDice: Int,
+                val nSides: Int,
+                var markedSides: MutableList<MutableList<Boolean>> = Array(nDice){ i -> Array(nSides){i -> false}.toMutableList() }.toMutableList(),
+                var cumReward: Double = 0.0
+                ){
+
+
+    val diceConfig = Array(nDice){ i -> Array(nSides){i -> i + 1}.toList() }.toList()
+
+    fun roll(): MutableList<MutableList<Boolean>> {
+        // For each dice generate a random integer between (including) 0 to nSides-1
+        var diceRollResults = mutableListOf<Int>()
+        for (d in markedSides) {
+            diceRollResults.add(Random.nextInt(0, nSides-1))
+            // Use -1 to represent unmarking
+        }
+
+
+        for (d in markedSides.indices) {
+            var rollInd = diceRollResults[d]
+            if (diceRollResults.filter{it == 1}.size > 1){
+                markedSides[d].set(rollInd, false)
+            } else {
+                markedSides[d].set(rollInd, !markedSides[d][rollInd])
+            }
+
+        }
+        return markedSides
+    }
+
+    fun cashOut(): Double {
+        for (d_ind in diceConfig.indices){
+            if (markedSides[d_ind] != Array(nSides){i -> false}.toMutableList() ){
+                var runningSum = 0.01
+                for (s_ind in 0 until diceConfig[d_ind].size){
+                    if (markedSides[d_ind][s_ind]){
+                        runningSum *= diceConfig[d_ind][s_ind]
```

Comment:
Wait why is this a running sum if we are multiplying? Also, if it's a runningProduct, then the initial value should be 1. If it's actually a runningSum, then the initial value should be 0. In neither case does 0.01 make sense unless that's how the MDP is defined, in which case, let's call it runningReward since it's neither a sum or a product per se.

Reply:
I changed it from `cumReward` to `instantReward` to avoid confusion. I made the initial values 0.01 (could potentially change) to let the the reward values better match the exploration constants. But ideally it should be 1 - which I've changed back.

fyi Sum can refer to mult or add (ie. geometric sum vs arithmetic sum).

Reply:
ja, should be product

Reply:
potential for alternate scaling factors, because depends on n dice and n sides, perhaps scaling the rewards elsewhere makes sense.